File: src/main/java/com/dv/telegram/config/ReplyWhenNoAnswer.java
package com.dv.telegram.config;

import lombok.Data;

@Data
public class ReplyWhenNoAnswer implements BotSetting<Boolean> {

    public static final String NAME = ReplyWhenNoAnswer.class.getSimpleName();

    private boolean value;

    @Override
    public String getName() {
        return NAME;
    }

    @Override
    public String getDescription() {
        return """
            *true* — Если бот не нашёл ответа, он выдаст ответ, определённый настройкой *NoAnswerReply*.

            *false* — Если бот не нашёл ответа, он не будет отвечать.
            """;
    }

    @Override
    public Boolean getValue() {
        return value;
    }

    @Override
    public void setValue(String value) {
        this.value = Boolean.parseBoolean(value);
    }
}

File: src/main/java/com/dv/telegram/config/ReplyWhenNoAnswer.kt
package com.dv.telegram.config

class ReplyWhenNoAnswer : BotSetting<Boolean> {

    companion object {
        // class simpleName cannot be const, see https://stackoverflow.com/questions/37182900/static-const-in-kotlin-from-java-class-name
        const val NAME: String = "ReplyWhenNoAnswer"
    }

    private var value = false

    override val name = NAME

    override val description = """
        *${true}* — Если бот не нашёл ответа, он выдаст ответ, определённый настройкой *${NoAnswerReply.NAME}*.

        *${false}* — Если бот не нашёл ответа, он не будет отвечать.
        """.trimIndent()

    override fun getValue() = value

    override fun setValue(value: String?) {
        this.value = value.toBoolean()
    }
}
